react/ProductCustomizer.js

```diff
@@ -1,54 +1,180 @@
+import PropTypes from 'prop-types'
 import React, { Component } from 'react'
 import Modal from 'vtex.styleguide/Modal'
 import Button from 'vtex.styleguide/Button'
 
 import './global.css'
-import AttachmentsList from './components/AttachmentsList'
+import schema from './__mock__/index.json'
+import AddToCart from './components/Buttons/AddToCart'
+import VariationList from './components/Variation/List'
+import ChangeToppings from './components/Buttons/ChangeToppings'
+import ProductCustomizerService from './utils/ProductCustomizerService'
 
 const isMobile = window.__RUNTIME__.hints.mobile
 
 class ProductCustomizer extends Component {
+  static propTypes = {
+    enableChangeToppings: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    enableChangeToppings: false,
+  }
+
+  state = {
+    total: 0,
+    variations: [],
+    isModalOpen: false,
+    selectedVariations: [],
+    selectionsValidated: false,
+  }
+
+  /**
+  * constructor
+  * Initilize the parent class contructor and parse the data from Schema
+  * @return void
+  */
   constructor() {
     super()
-    this.state = { isModalOpen: false }
-    this.handleOpenModal = this.handleOpenModal.bind(this)
-    this.handleCloseModal = this.handleCloseModal.bind(this)
+
+    const service = new ProductCustomizerService(schema)
+
+    this.state.variations = service.serializeData()
   }
 
-  handleOpenModal() {
+  /**
+  * onHandleRemovePreviousSelectedVariation
+  * Remove the current value from variations list and adds the new one.
+  * @param object variationToRemove
+  * @return bool|void
+  */
+  onHandleRemovePreviousSelectedVariation = (variationToRemove) => {
+    const selectedVariations = this.state.selectedVariations
+
+    const updatedSelectedArray = selectedVariations.filter((currentVariation) => {
+      return variationToRemove === undefined || (variationToRemove.Id !== currentVariation.Id)
+    })
+
+    this.setState({ selectedVariations: updatedSelectedArray })
+  }
+
+  /**
+  * onHandleAddSelectedVariations
+  * Add a new selected variation to array then updates total amount.
+  * @param object variation
+  * @return void
+  */
+  onHandleAddAndCalculateSelectedVariations = (variation) => {
+    const selectedVariations = this.state.selectedVariations
```

Comment:
In this case I think that should be better to use destructuring to retrieve the `selectedVariations` attribute.